commands/cli/parse.go

```diff
@@ -335,8 +340,16 @@ func appendStdinAsString(args []string, stdin *os.File) ([]string, *os.File, err
 func appendFile(args []files.File, inputs []string, argDef *cmds.Argument, recursive bool) ([]files.File, []string, error) {
 	path := inputs[0]
 
-	file, err := os.Open(path)
+	file, err := os.OpenFile(path, (os.O_RDONLY | syscall.O_NOFOLLOW | syscall.O_NONBLOCK), 0)
```

Comment:
lets just call stat before we call open. It might introduce a slight race, but its better than breaking our cross platformability with importing syscall.